contracts/src/BoundlessMarket.sol

```diff
@@ -566,9 +573,12 @@ contract BoundlessMarket is
 
         requestLocks[id].setProverPaidAfterLockDeadline(assessorProver);
         if (MARKET_FEE_BPS > 0) {
-            price = _applyMarketFee(price);
+            finalPrice = _applyMarketFee(finalPrice);
+        }
+        accounts[assessorProver].balance += finalPrice;
+        if (partialPayment) {
```

Comment:
I _think_ if `finalPrice != price`, it must be a partial payment, so could drop this variable